server/src/main/java/io/druid/server/listener/announcer/ListeningAnnouncerConfig.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.listener.announcer;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import io.druid.server.initialization.ZkPathsConfig;
+import org.apache.curator.utils.ZKPaths;
+
+/**
+ * Even though we provide the mechanism to get zk paths here, we do NOT handle announcing and unannouncing in this module.
+ * The reason is that it is not appropriate to force a global announce/unannounce since individual listeners may have
+ * different lifecycles.
+ */
+public class ListeningAnnouncerConfig
+{
+  @JacksonInject
+  private final ZkPathsConfig zkPathsConfig;
+  @JsonProperty("listenersPath")
```

Comment:
i do agree with keeping it as part of core, but thought the path can be part of `ListenerResourceAnnouncer` anyway no big deal.